File: sdks/typescript/src/resources/admin/backups.ts
import type { RequestClient } from "../../core.js";
import type { BackupCreateBody, BackupJob, BackupJobListResponse } from "../../types/index.js";
import { pagination } from "./_shared.js";

export class AdminBackupsResource {
  constructor(private readonly _client: RequestClient) {}

  list(options: { limit?: number; offset?: number } = {}): Promise<BackupJobListResponse> {
    return this._client._request("GET", "/v1/admin/backups", { params: pagination(options) });
  }

  get(backupId: string): Promise<BackupJob> {
    return this._client._request("GET", `/v1/admin/backups/${encodeURIComponent(backupId)}`);
  }

  create(body: BackupCreateBody = {}): Promise<BackupJob> {
    return this._client._request("POST", "/v1/admin/backups", {
      json: { label: body.label ?? "" },
    });
  }
}

File: sdks/python/src/bigrag/resources/admin/backups.py
from __future__ import annotations

from typing import TYPE_CHECKING
from urllib.parse import quote

from bigrag.resources.admin._shared import _pagination
from bigrag.types.admin import (
    BackupCreateBody,
    BackupJob,
    BackupJobListResponse,
)

if TYPE_CHECKING:
    from bigrag._core import BigRAGCore


class AdminBackupsResource:
    def __init__(self, client: BigRAGCore) -> None:
        self._client = client

    async def list(
        self, *, limit: int | None = None, offset: int | None = None
    ) -> BackupJobListResponse:
        params = _pagination(limit=limit, offset=offset)
        return await self._client._request("GET", "/v1/admin/backups", params=params)

    async def get(self, backup_id: str) -> BackupJob:
        return await self._client._request("GET", f"/v1/admin/backups/{quote(backup_id, safe='')}")

    async def create(self, body: BackupCreateBody | None = None) -> BackupJob:
        label = (body or {}).get("label", "")
        return await self._client._request("POST", "/v1/admin/backups", json={"label": label})
